### PR TITLE
Allow string resources without property or S.R.Extensions ref

### LINKED ISSUE #4474: Preserialized resources error fires when not required
See https://github.com/dotnet/corefx/pull/38964#discussion_r297945596

To reproduce, exclude System.Resources.Extensions.dll (either through DisableImplicitFrameworkReferences and manually adding references, or subtracting it with a target) from a netcoreapp3.0 project, add a resource with only string resources and build.

Expect: No Error

Actual: Error about missing S.R.E

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #4474

### CODE REVIEW

File: src/Tasks/GenerateResource.cs [hunk 1]
```diff
@@ -3393,6 +3370,67 @@ private void WriteResources(ReaderInfo reader, String filename)
...
+                {
+                    _haveSystemResourcesExtensionsReference = false;
+                    return false;
+                }
+
+                PopulateAssemblyNames();
+
+                foreach (var assemblyName in _assemblyNames)
+                {
+                    if (assemblyName.Name == "System.Resources.Extensions")
```
Comment (dsplaisted) [inline src/Tasks/GenerateResource.cs]: Should this be case insensitive?

File: src/Tasks/Resources/Strings.resx [hunk 2]
```diff
@@ -1127,6 +1127,10 @@
     <value>MSB3822: Preserialized resources require the System.Resources.Extensions assembly at runtime, but it was not found in this project's references.</value>
     <comment>{StrBegin="MSB3822: "}</comment>
   </data>
+  <data name="GenerateResource.PreserializedResourcesRequiresProperty">
+    <value>MSB3823: Preserialized resources require the property GenerateResourceUsePreserializedResources to be set to true.</value>
```
Comment (dsplaisted) [inline src/Tasks/Resources/Strings.resx]: Are people going to understand what these messages mean?  Are they going to know that they are using preserialized resources?

File: src/Tasks/GenerateResource.cs (hunk 1 above)
Comment (nguerrera) [reply src/Tasks/GenerateResource.cs]: Pre-existing, but fixed in 43186dc

File: src/Tasks/Resources/Strings.resx (hunk 2 above)
Comment (nguerrera) [reply src/Tasks/Resources/Strings.resx]: Also pre-existing (elsewhere), but fixed in d5b6ccf

**Comment by rainersigwald** [comment]
Belated LGTM, thanks!

**Pull request merged**